### ISSUE TITLE
Issues found testing with TalkBack screenreader

### ISSUE THREAD

**Issue by jancborchardt** [nextcloud/android#2780]
I did some quick testing with the [TalkBack screenreader](https://support.google.com/accessibility/android/answer/6283677) which is included in the Android accessibility settings and here’s some basic issues I found:

- Top left menu button just reads out "Open" – should be "Open sidebar"
- The "+" menu FAB is spoken as "unlabeled button", should be "Add or upload"
- In the file list, "thumbnail" is spoken, shouldn't be
- In the file list, "more icon menu" should be simply "More menu"
- ~~In the sidebar, the active menu item has "checked" prepended ("checked All files"), probably shouldn’t have~~
- ~~In the file list, it reads out "4 MB, 2 days ago", but should either say "4 MB, modified 2 days ago", or not read it at all?~~

And from @henningoschwald:

- The content descriptions currently are quite visual. For example "fingerprint icon", "shared icon" or "activity icon". I would like to suggest to strip the "icon" in this case.
- In the file list, the content description (currently "shared icon") should indicate whether a file or folder is currently shared. In case the item is not shared, the content description should be "share", if the item is shared, it should be "shared".

What do you think @nextcloud/accessibility @henningoschwald? cc @nextcloud/android

**Comment by AndyScherzinger** [comment]
Sounds good to me and can be achieved (easily) except for the following issues:

> In the sidebar, the active menu item has "checked" prepended ("checked All files"), probably shouldn’t have

The menu item's state is "checked" which is why TalkBack seems to say "checked" so it is an object state itself, nothing we can influence.

> In the file list, it reads out "4 MB, 2 days ago", but should either say "4 MB, modified 2 days ago", or not read it at all?

Would then probably mean that we need to change the text itself since TalkBack is simply reading out the textview. I did find a way to silence TalkBack (https://stackoverflow.com/questions/15723797/android-prevent-talkback-from-announcing-textview-title-aloud) while that might be a bit of a workaround/hack besides the fact that one might want to have this information, so I'd rather go for changing the way the info is read out loud then blocking.

**Comment by henningoschwald** [comment]
Many thanks @jancborchardt for doing these tests!

Here are my comments:
* "Top left menu button just reads out "Open" – should be "Open sidebar"":
Agreed.
* "The "+" menu FAB is spoken as "unlabeled button", should be "Add new"":
Yes, the FAB should be definitely labeled. As the menu also provides uploading functions, I propose "add/upload" or similar as content description.
* The "checked" prepending the active item in the main menu can be kept imho, as it gives information about the selected view.
* "In the file list, "thumbnail" is spoken, shouldn't be":
Agreed. This is not a relevant information and just clutters up the spoken output.
* "In the file list, it reads out "4 MB, 2 days ago", but should either say "4 MB, modified 2 days ago", or not read it at all?": Imho this is fine as it is, since it is unambiguous. Speaking "modified" would also be okay, since this is a short word, but generally keep in mind all text has to be spoken by the TTS which takes the more time the longer the text is.
* "    In the file list, "share icon menu" and "more icon menu" don't say for which file/folder, should be "Share filename.png" and "Settings for foldername":
Imho this is also fine as it is because the buttons are located next to the file or folder. At least for me it was never an issue to assign the buttons to the item to which they belong, neither by navigating by the left or right swipe gestures nor when moving around with my finger. Im not sure whether there exists a best practice for this case, but I prefer speech output which is as less chatty as possible while not missing relevant information.

I also have some additions:
* The content descriptions currently are quite visual. For example "fingerprint icon", "shared icon" or "activity icon". I would like to suggest to strip the "icon" in this case.
* In the file list, the content description (currently "shared icon") should indicate whether a file or folder is currently shared. In case the item is not shared, the content description should be "share", if the item is shared, it should be "shared".

Cheers and thanks again!

**Comment by jancborchardt** [comment]
Good stuff, thanks for the feedback @henningoschwald @AndyScherzinger! :) I adjusted the original issue, striked out the entries about the "Checked" and "4 MB, 2 days ago", and added both suggestions by @henningoschwald. 

@nextcloud/android if anyone of you wants to dive in here, it’s some nice small issues to fix for our accessibility initiative. :)

**Comment by jancborchardt** [comment]
(Also @henningoschwald would it be ok to add you to the Nextcloud accessibility team? This would mean you will receive notifications about accessibility issues or pull requests where that team is mentioned, for feedback or review. :) You can set to only receive them in the Github web interface, so your email inbox is not so cluttered.)

**Comment by henningoschwald** [comment]
Sure @jancborchardt, please add me to the accessibility team. I am happy to help.

**Comment by tyrylu** [comment]
I just wanted to thank you for your interest in this topic, it is not as common as it should and as i have no Android device, i could not do the testing myself.

**Comment by AndyScherzinger** [comment]
PR #2847 fixes all issues mentioned in this issue :heart: Thanks everyone for the very valuable feedback. This is highly appreciated. If you find anything else missing regarding talkback support and accessibility don't hesitate. We are eager to push this forward :)

**Comment by AndyScherzinger** [comment]
PR has been merged so the fixes will ship with the next stable release and will also be part of the next nightly build for testing :tada: 

Thanks everyone for the feedback and for taking the time to test it and comment on this ticket :heart:

**Comment by jancborchardt** [comment]
Awesome @AndyScherzinger, thanks a lot! :)

**Comment by AndyScherzinger** [comment]
My pleasure ❤️
